pkg/pipeline/lifecycle.go

```diff
@@ -623,6 +637,7 @@ func (s *Service) runPipeline(ctx context.Context, pl *Instance) error {
 			Str(log.PipelineIDField, pl.ID).
 			Msg("pipeline stopped")
 
+		s.notify(pl.ID, pl.ProvisionedBy, err)
```

Comment:
Maybe the logger statement confused you, but at this point err could be `nil`. We should only notify if `err` is not `nil`. The logger statement works, because it chooses the level based on the error - if `err` is `nil` it's INFO, otherwise it's ERROR.

Reply:
I think I skimmed through the code and saw the two cases... I think I'll build the nil check into `notify`, it's easier to use.